tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -78,6 +117,17 @@ void RFriendInfo::AddFriend(const std::vector<std::pair<std::string, std::string
    }
 }
 
+///////////////////////////////////////////////////////////////////////////////
+/// Get all the top-level branches names, including the ones of the friend trees
+std::vector<std::string> GetTopLevelBranchNames(TTree &t)
+{
+   std::set<std::string> bNamesSet;
```

Comment:
I know this was already there, but I think the safe default, for performance in case of a large number of branches, is `std::unordered_set` rather than `std::set`.